pkg/plugins/golang/options.go

```diff
@@ -0,0 +1,256 @@
+func (opts Options) Validate() error {
+	// Check that the required flags did not get a flag as their value
+	// We can safely look for a '-' as the first char as none of the fields accepts it
+	// NOTE: We must do this for all the required flags first or we may output the wrong
+	// error as flags may seem to be missing because Cobra assigned them to another flag.
+	if strings.HasPrefix(opts.Group, "-") {
+		return fmt.Errorf(groupPresent)
+	}
+	if strings.HasPrefix(opts.Version, "-") {
+		return fmt.Errorf(versionPresent)
+	}
+	if strings.HasPrefix(opts.Kind, "-") {
+		return fmt.Errorf(kindPresent)
+	}
+	// Now we can check that all the required flags are not empty
+	if len(opts.Version) == 0 {
+		return fmt.Errorf(versionRequired)
+	}
+	if len(opts.Kind) == 0 {
+		return fmt.Errorf(kindRequired)
+	}
+
+	// Check if the qualified group has a valid DNS1123 subdomain value
+	if err := validation.IsDNS1123Subdomain(opts.QualifiedGroup()); err != nil {
+		return fmt.Errorf("either group or domain is invalid: (%v)", err)
+	}
+
+	// Check if the version follows the valid pattern
+	if !versionRegex.MatchString(opts.Version) {
+		return fmt.Errorf("version must match %s (was %s)", versionPattern, opts.Version)
+	}
+
+	validationErrors := []string{}
+
+	// Require kind to start with an uppercase character
+	if string(opts.Kind[0]) == strings.ToLower(string(opts.Kind[0])) {
+		validationErrors = append(validationErrors, "kind must start with an uppercase character")
+	}
+
+	validationErrors = append(validationErrors, validation.IsDNS1035Label(strings.ToLower(opts.Kind))...)
+
+	if len(validationErrors) != 0 {
+		return fmt.Errorf("invalid Kind: %#v", validationErrors)
+	}
+
+	// TODO: validate plural strings if provided
+
+	// Ensure apiVersions for k8s types are empty or valid.
+	for typ, apiVersion := range map[string]string{
+		"CRD":     opts.CRDVersion,
+		"Webhook": opts.WebhookVersion,
+	} {
+		switch apiVersion {
+		case "", "v1", "v1beta1":
+		default:
+			return fmt.Errorf("%s version must be one of: v1, v1beta1", typ)
+		}
+	}
+
+	return nil
+}
+
+// QualifiedGroup returns the fully qualified group name with the available information.
+func (opts Options) QualifiedGroup() string {
+	switch "" {
+	case opts.Domain:
+		return opts.Group
+	case opts.Group:
+		return opts.Domain
+	default:
+		return fmt.Sprintf("%s.%s", opts.Group, opts.Domain)
+	}
+}
+
+// GVK returns the GVK identifier of a resource.
+func (opts Options) GVK() resource.GVK {
+	return resource.GVK{
+		Group:   opts.QualifiedGroup(),
+		Version: opts.Version,
+		Kind:    opts.Kind,
+	}
+}
+
+// NewResource creates a new resource from the options
+func (opts Options) NewResource(c *config.Config) *resource.Resource {
+	res := &resource.Resource{
+		Group:      opts.Group,
+		Domain:     opts.Domain,
+		Version:    opts.Version,
+		Kind:       opts.Kind,
+		Path:       resource.LocalPath(c.Repo, opts.Group, opts.Version, c.MultiGroup),
```

Comment:
It shows fit better in the config where we would return the default endpoint for the Group and Version informed according to one config project. See: https://github.com/kubernetes-sigs/kubebuilder/pull/1696/files#diff-5e3c06a663e77e82b3da9e45c08d6b209590130fcf612ebf2af9343e1dd71f02R389

Reply:
Semantically, `path` (or `endpoint`) has nothing to do with the config file, it has to do with the resource, so making it a method of the config file makes no sense.

Reply:
The Path/Endpoint is for the specific PROJECT. The Config is the object which represents the PROJECT. 
So, we are liking here for What is  Path/Endpoint for the Group and Version on the specific PROJECT/Config.